flink-runtime/src/test/java/org/apache/flink/runtime/testutils/CommonTestUtils.java

```diff
@@ -359,6 +364,53 @@ public static void waitForSubtasksToFinish(
                 Deadline.fromNow(Duration.of(1, ChronoUnit.MINUTES)));
     }
 
+    /** Wait for at least one successful checkpoint. */
+    public static void waitForCheckpoint(JobID jobID, MiniCluster miniCluster, Deadline timeout)
+            throws Exception, FlinkJobNotFoundException {
+        waitUntilCondition(
+                () ->
+                        Optional.ofNullable(
+                                        miniCluster
+                                                .getExecutionGraph(jobID)
+                                                .get()
+                                                .getCheckpointStatsSnapshot())
+                                .filter(st -> st.getCounts().getNumberOfCompletedCheckpoints() > 0)
+                                .isPresent(),
+                timeout);
+    }
+
+    /**
+     * @return the path as {@link java.net.URI} to the latest checkpoint.
+     * @throws FlinkJobNotFoundException if job not found
+     * @throws NoSuchElementException if no checkpoint is created yet
+     */
+    public static String getLatestCompletedCheckpointPath(JobID jobID, MiniCluster cluster)
+            throws FlinkJobNotFoundException, NoSuchElementException, ExecutionException,
+                    InterruptedException {
+        return getLatestCompletedCheckpointPathMaybe(jobID, cluster)
+                .<NoSuchElementException>orElseThrow(
+                        () -> {
+                            throw new NoSuchElementException("No checkpoint was created yet");
+                        });
+    }
```

Comment:
I'd remove this method, the Optional one is easy enough to use, which I'd rename to `getLatestCompletedCheckpointPath`.